content/guides/concepts/hashes.md

```diff
@@ -0,0 +1,59 @@
+---
+title: "Hashes"
+menu:
+    guides:
+        parent: concepts
+---
+
+Hashes are functions that take some arbitrary input and returns a fixed-length value. The particular value depends on the given hash algorithm in use, such as [SHA-1](https://en.wikipedia.org/wiki/SHA-1) (used by Git), [SHA-256](https://en.wikipedia.org/wiki/SHA-2), or [BLAKE2](https://en.wikipedia.org/wiki/BLAKE_(hash_function)#BLAKE2), but a given hash algorithm always returns the same value for a given input. Have a look at the [full list of hash functions](https://en.wikipedia.org/wiki/List_of_hash_functions) for more.
+
+As an example, the input:
+
+```
+Hello world
+```
+
+would be represented by **SHA-1** as:
+
+```
+0x7B502C3A1F48C8609AE212CDFB639DEE39673F5E
+```
+
+However, the exact same input generates the following output using **SHA-256**:
+
+```
+0x64EC88CA00B268E5BA1A35678A1B5316D212F4F366B2477232534A8AECA37F3C
+```
+
+Notice that the second hash is longer than the first one. This is because SHA-1 creates a 160 bit hash, while SHA-256 creates a 256 bit hash. Also, the prepended `0x` is just an indicator that tells us that the following hash is represented as a base 16 (or hexadecimal) number.
+
+Hashes can be represented in different bases (`base2`, `base16`, `base32`, etc.). In fact, IPFS makes use of that as part of its [Content Identifiers](cid.md) and supports mulitiple base representations at the same time, using the [Multibase](https://github.com/multiformats/multibase) protocol.
+
+For example, the SHA-256 hash of "Hello World" from above can be represented as base 32 as:
+
+```
+mtwirsqawjuoloq2gvtyug2tc3jbf5htm2zeo4rsknfiv3fdp46a
+```
+
+An easy way to play around with these in JavaScript is to use the [multibase package](https://github.com/multiformats/js-multibase):
+
+```
+const crypto = require('crypto');
+const multibase = require('multibase');
```

Comment:
@Mr0grog notice that I've added the require statements here just for completeness' sake

Reply:
OK. Sorry, I now realize I was really unclear with my earlier comment. (In my head, the horizontal rule was ending the main comment and starting a different discussion. Reading it now I see how that was not obvious.)

I’m not sure this whole bit with the code sample fits in really well here — we jump from some basic examples of hashes and their formatting to code that uses a whole bunch concepts we haven’t covered in the doc. That might be a little much.